src/stylesheets/core/_functions.scss

```diff
@@ -1698,263 +1814,553 @@ system z-index
-        }
-
-        // set family and grade for "accent" families, since their family includes the split character
-        @if $custom-family == "accent" {
-          $custom-family: $custom-family + "-" + nth($custom-split, 2);
-          $custom-grade: if(
-            length($custom-split) == 3,
-            nth($custom-split, 3),
-            "default"
-          );
-        } @else {
-          $custom-grade: if(
-            length($custom-split) == 2,
-            nth($custom-split, 2),
-            "default"
-          );
-        }
-
-        //@debug "custom family: " + $custom-family;
-        //@debug "custom grade: " + $custom-grade;
-
-        $custom-family-lighter: $custom-family;
-        $custom-family-darker: $custom-family;
-        $custom-grade-index: index($uswds-color-theme-grades, $custom-grade);
-
-        // If it's the lightest grade, use "white" for the lighter family
-        @if $custom-grade-index == 1 {
-          $custom-family-lighter: "white";
-        } @else {
-          $custom-grade-lighter: nth(
-            $uswds-color-theme-grades,
-            ($custom-grade-index - 1)
-          );
-        }
-        //@debug "lighter grade: " + $custom-grade-lighter;
-        // If it's the darkest grade, use "black" for the lighter family
-        @if $custom-grade-index == length($uswds-color-theme-grades) {
-          $custom-family-darker: "black";
-        } @else {
-          $custom-grade-darker: nth(
-            $uswds-color-theme-grades,
-            ($custom-grade-index + 1)
-          );
-        }
-        //@debug "darker grade: " + $custom-grade-darker;
-
-        // If any calculated grade is "default", don't output the grade
-        $custom-grade-darker: if(
-          $custom-grade-darker == "default",
-          false,
-          $custom-grade-darker
-        );
-        $custom-grade-lighter: if(
-          $custom-grade-lighter == "default",
-          false,
-          $custom-grade-lighter
-        );
-
-        // Build the custom lighter and darker tokens
-        $token-darker: if(
-          $custom-grade-darker,
-          $custom-family + "-" + $custom-grade-darker,
-          $custom-family-darker
-        );
-        $token-lighter: if(
-          $custom-grade-lighter,
-          $custom-family + "-" + $custom-grade-lighter,
-          $custom-family-lighter
-        );
-      } @else {
-        //@debug "not custom";
-        $decomposed: decompose($color-token);
-      }
+color-token-family("red-50v")
+> "red"
 
-      @if $link-grade == 0 {
-        @warn 'Tokens with grades less than 10 (including "white") aren\'t valid link color tokens, since they have no lighter hover states.';
-      } @else if $link-grade == 100 {
-        @warn '"black" isn\'t a valid link color token, since it has no darker hover state.';
-      }
+color-token-variant(("red", 50, "vivid"))
+> "red"
+----------------------------------------
+*/
+
+@function color-token-family($color-token) {
```

Comment:
This (and `color-token-family()` and `color-token-grade()`) can accept a token or a split. Split is faster!